File: Workspace/Personal Training/JSW/Baekjoon/28278/solution.py
class Node:
 def __init__(self, data, next):
  self.data = data
  self.next = next
class stack:
 def __init__(self):
  self.top = Node(None, None)
  self.size = 0

 def push(self, x):
  temp = Node(x, self.top)
  self.top = temp
  self.size += 1

 def pop(self):
  if self.is_empty():
   return -1
  temp = self.top.data
  self.top = self.top.next
  self.size -= 1
  return temp
 
 def peek(self):
  if self.is_empty():
   return -1
  return self.top.data
 
 def __len__(self):
  return self.size
 
 def is_empty(self):
  return self.size == 0
  


stack  = stack()
N = int(input())
output = []
for i in range(N):
 line = input().split()
 if line[0] == '1':
  stack.push(int(line[1]))
 elif line[0] == '2':
  output.append(stack.pop())
 elif line[0] == '3':
  output.append(len(stack))
 elif line[0] == '4':
  output.append(1 if stack.is_empty() else 0)
 else:
  output.append(stack.peek())

print(*output, sep='\n')

File: Workspace/Personal Training/JSW/Baekjoon/28278/solution.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.IO;


public class Program
{
    public static void Main()
    {
        MyStack<string?> stack = new ();
        StringBuilder sb = new StringBuilder();
        using (var sr = new StreamReader(Console.OpenStandardInput())) 
        {
            int N = int.Parse(sr.ReadLine());

            for (int i = 0; i < N; i++)
            {
                var line = sr.ReadLine().Split(' ');
                switch (line[0])
                {
                    case "1":
                        stack.Push(line[1]);
                        break;
                    case "2":
                        sb.AppendLine(stack.Pop() ?? "-1");
                        break;
                    case "3":
                        sb.AppendLine(stack.Count.ToString());
                        break;
                    case "4":
                        sb.AppendLine(stack.IsEmpty() ? "1" : "0");
                        break;
                    case "5":
                        sb.AppendLine(stack.Peek() ?? "-1");
                        break;
                }
            }
        }
        

        Console.Write(sb);
    }
}

public class Node<T>
{
    public T Value { get; set; }
    public Node<T>? Next { get; set; }
    public Node(T value, Node<T>? next = null)
    {
        Value = value;
        Next = next;
    }
}

public class MyStack<T>
{
    private Node<T>? Top { get; set; }
    public int Count { get; private set; }

    public MyStack()
    {
        Top = new Node<T>(default, null);
        Count = 0;
    }

    public void Push(T value)
    {
        Top = new Node<T>(value, Top);
        Count++;
    }

    public bool IsEmpty()
    {
        return Count == 0;
    }

    public T? Pop()
    {
        if (IsEmpty()) return default;
        T value = Top!.Value;
        Top = Top.Next;
        Count--;
        return value;
    }

    public T? Peek()
    {
        return IsEmpty() ? default : Top!.Value;
    }
}
